Lib/ast.py

```diff
@@ -80,7 +80,7 @@ def _convert(node):
         elif isinstance(node, BinOp) and isinstance(node.op, (Add, Sub)):
             left = _convert_signed_num(node.left)
             right = _convert_num(node.right)
-            if isinstance(left, (int, float)) and isinstance(right, complex):
+            if type(left) in (int, float) and type(right) is complex:
```

Comment:
I don't think it affects much but looks better to me
```suggestion
            if type(left) in {int, float} and type(right) is complex:
```

Reply:
It is faster to create a tuple than a set.